api/shared/src/main/scala/dagr/api/DagrApi.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package dagr.api
+
+import dagr.api.DagrApi.TaskId
+import dagr.api.models._
+
+/*
+object DagrApiConfiguration {
+  object Keys {
+    val ApiVersion = "dagr.api.version"
+    val ApiRoot    = "dagr.api.root"
+  }
+}
+*/
+
+/*
+trait DagrApiConfiguration extends Configuration {
+  lazy val version: String = optionallyConfigure[String](DagrApiConfiguration.Keys.ApiVersion).getOrElse("v1")
+  lazy val root: String    = optionallyConfigure[String](DagrApiConfiguration.Keys.ApiRoot).getOrElse("api")
+}
+*/
+
+trait DagrApiConfiguration {
+  lazy val version: String = "v1"
+  lazy val root: String    = "api"
+}
+
+object DagrApi {
+  type TaskId = BigInt
+
+  // *** IMPORTANT *** this must be the fully-qualified path to the DagrApi trait!
+  val prefixSegments: List[String] = List("dagr", "api", "DagrApi")
+}
+
+// NB: this trait cannot extend anything, as required by autowire
+trait DagrApi {
```

Comment:
This is implemented on the server-side and called by the scala-js side (via [autowire](https://github.com/lihaoyi/autowire))